rafthttp/pipeline.go

```diff
@@ -128,31 +132,43 @@ func (p *pipeline) post(data []byte) (err error) {
 		}
 	}()
 
-	resp, err := p.tr.pipelineRt.RoundTrip(req)
-	done <- struct{}{}
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	resp.Body.Close()
-
-	err = checkPostResponse(resp, b, req, p.peerID)
-	if err != nil {
-		p.picker.unreachable(u)
-		// errMemberRemoved is a critical error since a removed member should
-		// always be stopped. So we use reportCriticalError to report it to errorc.
-		if err == errMemberRemoved {
-			reportCriticalError(err, p.errorc)
+	go func() {
```

Comment:
@xiang90 i went through the code and i found that the issue is with the test case. Earlier the request cancel was being handled by us manually, hence the roundTripperBlocker used for test case is checking for req.Cancel which used to come earlier from our manual implementation. Now that is not the case and since the roundTripperBlocker has no implementation for ctx.Done hence it was failing. 
Will correct the same and update the PR. Thanks!